### DISCUSSION TITLE
Regarding api key

### DISCUSSION THREAD

**Discussion by nihalgurjar47** [ory/.github#71] [Q&A] [answered]
I require an API key for SDK configuration. In your documentation, it is mentioned that the API key is only available via the ORY console, but my requirement is for self-hosting. Is there any way to generate an API key for self-hosting?

**Comment by vinckr** [comment, marked as answer]
Hello @nihalgurjar47 

Yes, there are many projects out there that help you implement something like API-keys for your APIs.

The Ory Network API key solution is not open source, as it is very specific to Ory Network and could not be used in self-hosted deploys 
Unfortunately this is not in the scope of the Ory open source project at the moment - although there are plans to build a solution for this in Ory sometime next year.
Feel [free to contact the team](https://www.ory.sh/contact/) if you are looking for commercial support here; Ory can help you implement this in your self-hosted Ory deploys as well.
